app/workers/process_content_change_and_generate_emails_worker.rb

```diff
@@ -0,0 +1,74 @@
+class ProcessContentChangeAndGenerateEmailsWorker < ProcessAndGenerateEmailsWorker
+  include Sidekiq::Worker
+
+  sidekiq_options queue: :process_and_generate_emailsg
+
+  def perform(content_change_id)
+    content_change = ContentChange.find(content_change_id)
+    return if content_change.processed?
+
+    begin
+      import_subscription_content(content_change)
+    rescue ActiveRecord::RecordNotUnique, ActiveRecord::RecordInvalid, ActiveRecord::StatementInvalid, ActiveRecord::InvalidForeignKey
+      GovukStatsd.increment("process_content_change_and_generate_emails_worker.rescheduled")
+      raise
+    end
+
+    begin
+      SubscribersForImmediateEmailQuery.call(content_change_id: content_change_id, message_id: nil).find_in_batches(batch_size: batch_size) do |group|
+        ActiveRecord::Base.transaction do
+          subscription_contents = UnprocessedSubscriptionContentsBySubscriberQuery.call(group.pluck(:id))
+          if subscription_contents.any?
+            update_content_change_cache(subscription_contents)
+            create_content_change_emails(group, subscription_contents)
+          end
+        end
+      end
+      queue_for_delivery
+      content_change.mark_processed!
+      queue_courtesy_email(content_change)
+    rescue ActiveRecord::RecordNotUnique, ActiveRecord::RecordInvalid, ActiveRecord::StatementInvalid, ActiveRecord::InvalidForeignKey
+      GovukStatsd.increment("process_content_change_and_generate_emails_worker.rescheduled")
```

Comment:
I'm not sure we get a lot from this statsd value, as if an exception outside of those listed raises an error the job would get rescheduled anyway so the stat would be inaccurate.

Reply:
True. From writing the tests I found that the errors listed occur when there is contention for records - so if you think there is value in recording errors arising from that we could keep it but otherwise you're right it's not especially useful

Reply:
Removed now 😄